lib/data/external/GCP/GcpApis/putTagging.js

```diff
@@ -0,0 +1,33 @@
+const async = require('async');
+const { errors } = require('arsenal');
+
+const { processTagSet } = require('../GcpUtils');
+
+function putObjectTagging(params, callback) {
+    if (!params.Tagging || !params.Tagging.TagSet) {
+        return callback(errors.MissingParameter);
+    }
+    const tagRes = processTagSet(params.Tagging.TagSet);
+    if (tagRes instanceof Error) {
+        return callback(tagRes);
+    }
+    return async.waterfall([
+        next => this.headObject({
+            Bucket: params.Bucket,
+            Key: params.Key,
+            VersionId: params.VersionId,
+        }, next),
+        (resObj, next) => {
+            const completeMD = Object.assign({}, resObj.Metadata, tagRes);
```

Comment:
What is the behavior if the object already has tags for the metadata? Should we be overwriting them?

Reply:
I tested the behavior of AWS's object tagging manually and have found that previous tags are overwritten on each put-object-tagging call.